src/utils/build.js

```diff
@@ -321,171 +315,158 @@ const build = (zipPath, sourceZipPath, wdir) => {
       });
     }
   }
+};
 
-  return ensureDir(tmpDir)
-    .then(() => ensureDir(constants.BUILD_DIR))
-    .then(() => {
-      startSpinner('Copying project to temp directory');
-
-      let filter;
-      if (process.env.SKIP_NPM_INSTALL) {
-        filter = dir => {
-          const isntBuild = dir.indexOf('.zip') === -1;
-          return isntBuild;
-        };
-      }
-      return copyDir(wdir, tmpDir, { filter });
-    })
-    .then(() => {
-      if (process.env.SKIP_NPM_INSTALL) {
-        return {};
-      }
-      endSpinner();
-      startSpinner('Installing project dependencies');
-      return runCommand('npm', ['install', '--production'], { cwd: tmpDir });
-    })
-    .then(output => {
-      // `npm install` may fail silently without returning a non-zero exit code, need to check further here
-      const corePath = path.join(
-        tmpDir,
-        'node_modules',
-        constants.PLATFORM_PACKAGE
-      );
-      if (!fs.existsSync(corePath)) {
-        throw new Error(
-          'Could not install dependencies properly. Error log:\n' +
-            output.stderr
-        );
-      }
-    })
-    .then(() => {
-      endSpinner();
-      startSpinner('Applying entry point file');
-      // TODO: should this routine for include exist elsewhere?
-      return readFile(
-        path.join(
-          tmpDir,
-          'node_modules',
-          constants.PLATFORM_PACKAGE,
-          'include',
-          'zapierwrapper.js'
-        )
-      ).then(zapierWrapperBuf =>
-        writeFile(`${tmpDir}/zapierwrapper.js`, zapierWrapperBuf.toString())
-      );
-    })
-    .then(() => {
-      endSpinner();
-      startSpinner('Building app definition.json');
-      return _appCommandZapierWrapper(tmpDir, { command: 'definition' });
-    })
-    .then(rawDefinition => {
-      return Promise.all([
-        writeFile(
-          `${tmpDir}/definition.json`,
-          prettyJSONstringify(rawDefinition.results)
-        ),
-        Promise.resolve(rawDefinition.results)
-      ]);
-    })
-    .then(([fileWriteError, rawDefinition]) => {
-      if (fileWriteError) {
-        if (global.argOpts.debug) {
-          console.log('\nFile Write Error:');
-          console.log(fileWriteError);
-          console.log('');
-        }
-        throw new Error(
-          `Unable to write ${tmpDir}/definition.json, please check file permissions!`
-        );
-      }
+const build = async (zipPath, sourceZipPath, wdir) => {
+  wdir = wdir || process.cwd();
+  zipPath = zipPath || constants.BUILD_PATH;
+  sourceZipPath = sourceZipPath || constants.SOURCE_PATH;
+  const osTmpDir = await fse.realpath(os.tmpdir());
```

Comment:
```suggestion
  const osTmpDir = fse.realpathSync(os.tmpdir());
```

This needs to be sync version as `fs.realpath` is a [callback-style](https://nodejs.org/docs/latest-v8.x/api/fs.html#fs_fs_realpath_path_options_callback) function.

Reply:
I think this one's ok, since it's using `fs-extra`, which makes everything into promises. 

![](https://zappy.zapier.com/E2F10243-4AEF-40B6-A64D-5CE013D6D268.png)

Reply:
👍 TIL!